sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -761,6 +761,25 @@ export class MessageSession extends LinkEntity<Receiver> {
     }
   }
 
+  /**
+   * To be called when connection is disconnected to gracefully close ongoing receive request.
+   * @param connectionError - The connection error if any.
+   */
+  async onDetached(connectionError: AmqpError | Error | undefined): Promise<void> {
```

Comment:
The changelog talks about SessionLockLostError, where are we ensuring that?

Reply:
Hmmm, you're right.
Only one of the cases in the tests throws that error in peekLock. Looking at how I can improve here.

Reply:
onDetached triggered by the disconnect event for sessions would now throw a session-lock-lost since that seems more appropriate.

Let me know how this update looks.